library/src/main/java/com/novoda/downloadmanager/LiteDownloadMigrationService.java

```diff
@@ -0,0 +1,124 @@
+package com.novoda.downloadmanager;
+
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Binder;
+import android.os.IBinder;
+import android.os.SystemClock;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class LiteDownloadMigrationService extends Service {
+
+    private Migrator v1ToV2Migrator;
+    private ExecutorService executor;
+    private IBinder binder;
+    private Migrator.Callback migrationCallback;
+    private String updateMessage;
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        super.onStartCommand(intent, flags, startId);
+
+        Log.d(getClass().getSimpleName(), "onStartCommand");
+
+        if (v1ToV2Migrator != null) {
+            executor.execute(new Runnable() {
+                @Override
+                public void run() {
+                    Log.d(getClass().getSimpleName(), "Begin Migration");
+                    v1ToV2Migrator.migrate();
+                }
+            });
+        }
+
+        return START_STICKY;
+    }
+
+    @Override
+    public void onCreate() {
+        Log.d(getClass().getSimpleName(), "onCreate");
+        executor = Executors.newSingleThreadExecutor();
+        binder = new MigrationDownloadServiceBinder();
+        v1ToV2Migrator = MigrationFactory.createVersionOneToVersionTwoMigrator(
+                getApplicationContext(),
+                getDatabasePath("downloads.db"),
+                migrationCallback()
+        );
+
+        super.onCreate();
+    }
+
+    private Migrator.Callback migrationCallback() {
+        return new Migrator.Callback() {
+            @Override
+            public void onUpdate(String message) {
+                updateMessage = message;
+                if (migrationCallback != null) {
+                    migrationCallback.onUpdate(message);
+                }
+            }
+        };
+    }
+
+    @Nullable
+    @Override
+    public IBinder onBind(Intent intent) {
+        return binder;
+    }
+
+    @Override
+    public void onTaskRemoved(Intent rootIntent) {
```

Comment:
This is executed whenever a user swipes the app from recents. We also need to handle the case when the system kills a service / app.